### ISSUE TITLE
about using coc_medium

### ISSUE THREAD

**Issue by lai-pf** [ma-xu/Context-Cluster#2]
I try to run validate.py with coc_medium, it can not load the checkpoint. 
error message like these.
![image](https://user-images.githubusercontent.com/32856760/223297451-fd6569a2-cf6c-4122-a5fe-551d0d509279.png)

**Comment by myeyeget** [comment]
A few changes seem to be required in the code

In models/context_cluster.py change 

    Cluster class:  self.f --> self.fc1     (rename)
                          self.proj --> self.fc2    (rename)
                          self.v --> self.fc_v     (rename)

    Mlp class:      self.fc1 --> nn.Conv2d(in_features, hidden_features, kernel_size=1)                     (change linear to conv2d)
                         self.fc2 --> nn.Conv2d(hidden_features, out_features, kernel_size=1)                  (change linear to conv2d)
                         also remove all permutes in forward()

In validate.py change

    change line 155 to
                   load_checkpoint(model, args.checkpoint, args.use_ema, strict=False)

 Although I'm not the author of this paper, these changes seemed to do the trick for me

**Comment by ma-xu** [comment]
@lai-pf @myeyeget Sry for the late reply.  
We slightly re-organized the codes for fast training and inference. @myeyeget 's solution seems good. 
You can also try the submission link here: https://anonymous.4open.science/r/ContextCluster/README.md

We will update the checkpoints soon. Thanks for your issue.

**Comment by ErlebnisW** [comment]
@ma-xu The submission link has expired. Could you please update it so that I can solve the mismatch problem?

**Comment by ma-xu** [comment]
@ErlebnisW @myeyeget @lai-pf 
Thanks for your patience. The updated checkpoints are updated, with slightly improvements. Mainly two updates: 1) fix a bug mask.sum to sim.sum; 2) conv1x1 to Linear implementation.

**Comment by ma-xu** [comment]
Closed no further discussions. Feel free to reopen it if any further questions.
